### PR TITLE
[CPU BF16] OneHot layer extended with bf16 support.

### PR DESCRIPTION
**Pull request by maxnick**
The OneHot layer has been extended with bf16 support mainly to fix some functional issues of enabling greedy mode.

### CODE REVIEW

**Review by alexey-varyzgin — APPROVED** [review]

**Comment by maxnick** [comment]
@yury-intel Could you please review?

**Review by yury-intel — APPROVED** [review]

**Comment by maxnick** [comment]
@yury-intel Thank you!
@dmitry-gorokhov Could you please review?

File: inference-engine/src/mkldnn_plugin/nodes/one_hot.cpp [hunk 1]
```diff
@@ -36,38 +41,41 @@ class OneHotImpl: public ExtLayerBase {
...
-            case Precision::FP32:
-                one_hot<float>(inputs[0], outputs[0]);
-                break;
-            case Precision::I32:
-                one_hot<int>(inputs[0], outputs[0]);
-                break;
-            default:
-                return GENERAL_ERROR;
+        OneHotContext ctx = {this, inputs[0], outputs[0], false};
+        OV_SWITCH(MKLDNNPlugin, OneHotExecute, ctx, std::tie(input_precision, output_precision),
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/one_hot.cpp:60]: As we discussed offline: using floating point precisions for indices input is not valid. Should be I32 only.

**Review by dmitry-gorokhov — CHANGES_REQUESTED** [review]

File: inference-engine/src/mkldnn_plugin/nodes/one_hot.cpp (hunk 1 above)
Comment (maxnick) [reply inference-engine/src/mkldnn_plugin/nodes/one_hot.cpp]: Fixed in 8a2d1b3. Also I8 output precision support was added.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**